### ISSUE TITLE
Cannot download audiobooks

### ISSUE THREAD

**Issue by pmyers389** [houtianze/libby-audiobook-exporter-browser-extension#1]
I would like to use this extension but it would not download any of the 4 books I tried. Case in point: The Four Winds by Kristin Hannah. I get the following message when I try to download: "Failed to retrieve the list, please refresh your tab (Ctrl + R)". Also the option "when you click the extension" does not work.

**Comment by houtianze** [comment]
Hi, thanks for the feedback. For some reason I'm still not clear, this extension seems to only work after a second reload (Ctrl + R), but for me, so far it works on all the audio books I tried. Can you try do a reload after the audibook page first fully loaded?

The "when you click the extension" one is something that I didn't implment at all, so I think that's the reason, but I think that would be a better place to show the extension page, and _maybe_ I will change to this way in the future.

**Comment by pmyers389** [comment]
I have 4 books I wanted to download. I was able to download the first book but not the others. Then I tried downloading the first book again but got the "failed to retrieve" message. I did Ctrl+R many times but it didn't help. Any other suggestions?

**Comment by houtianze** [comment]
I thought about it, is it that you opened 4 tabs at the same time? This extension is quite primary, and that may not work (I havn't confirmed this yet, but will look into this when I got some time in the future). At the time being, you can work around this by just opening one tab to download each audiobook. Once it's done, continue with others. I know, it's not that convinent, but till I found a solution, this is the only way I know of should work.

Btw, there is some feedback that all the files are always downloaded to the user's download folder, I will see if I can improve this on the next update.

**Comment by pmyers389** [comment]
No, I did not have 4 tabs open. I only try to download 1 book at a time. I did get this to work one time, but now I cannot download any books at all.

**Comment by pmyers389** [comment]
Download speed: 648 mbps, Download Latency: 153 ms. Here's a recommendation. If you haven't already, take a look at an extension on Firefox called Libby Download. I think you might find it interesting.

**Comment by houtianze** [comment]
Your network speed is very fast, and this shouldn't be an issue, and 30 second should be enough for timeout. Now I don't really have a clue what's causing this. Just to check that you are able to play those audio books in your chrome right? The only thing I can think of is trying to reinstall this extension, but I doubt it would really help.

Thanks for the Firefox pointer, I saw it just after I published this extension ... (Honestly, if I know it existed, I won't create this, it's much more polished than this one). Maybe I can compare with it to see why mine isn't working reliably.

**Comment by pmorgan111** [comment]
Well, it looks like it's working pretty good now. Maybe I was doing something wrong. I have no idea. Anyway, sorry to bother you.

**Comment by houtianze** [comment]
No problem and thanks for the update.

**Comment by pmyers389** [comment]
Hello,
Just a heads up. I did find out what was causing the extension to fail with "Failed to retrieve list......". If the user selects option "when you click the extension" the problem starts. It looks like "Site access" is changed in the extension details and the extension fails. Maybe there is a way for you to fix this so that dumb users like me don't get this problem?

**Comment by houtianze** [comment]
Thanks @pmyers389. This sounds indeed a bug, let me check how this can be fixed.

**Comment by pmyers389** [comment]
That would be great!

**Comment by sensibleroommate** [comment]
Hi, I installed the extension but don't see where I could find the option to download?
I open the audiobook in my browser and start to listen to it, refresh the page a few times and click on the extension, but there is no export or download or anything like that. Below are the options I see: How do I resolve this and download the audiobook?
![Screen Shot 2023-09-21 at 1 45 39 AM](https://github.com/houtianze/libby-audiobook-exporter-browser-extension/assets/145643032/45a52aa4-1f73-49e0-9a30-6f9462562b9d)

**Comment by pmyers389** [comment]
After you load the book into libbyapp.com do not start the audio, just refresh the screen. The extension will start automatically. When the extension is ready to download the audiobook the extension heading will display "download | toggle list". Click on the Export button and the book will be downloaded.

**Comment by sensibleroommate** [comment]
Thanks it worked! How do I get to the libbyapp.com from my library? I was opening the audiobook in my library and didn't see this option

**Comment by pmyers389** [comment]
If you accessed your library on a PC, I suspect when you opened the audiobook it took you to libbyapp.com automatically.  If you accessed your library with something other than a PC (or maybe Chrome on an Apple device) I cannot be helpful. Anyway, whatever you did seems to work, so that would be good enough for me.

**Comment by sensibleroommate** [comment]
Thanks, that seemed to work initially but now it just takes me to libbyapp.com/shelf automatically and when I open the audiobook, I don’t see an option, even after reloading the page to download or export anything it just opens the audiobook to listen to it. Even if I type libbyapp.com, it automatically adds the /shelf part and takes me to my audio shelf On Sep 21, 2023, at 7:23 PM, pmyers389 ***@***.***> wrote:﻿
If you accessed your library on a PC, I suspect when you opened the audiobook it took you to libbyapp.com automatically.  If you accessed your library with something other than a PC (or maybe Chrome on an Apple device) I cannot be helpful. Anyway, whatever you did seems to work, so that would be good enough for me.

**Comment by sensibleroommate** [comment]
Thanks, that seemed to work initially but now it just takes me to libbyapp.com/shelf automatically and when I open the audiobook, I don’t see an option, even after reloading the page to download or export anything it just opens the audiobook to listen to it. Even if I type libbyapp.com, it automatically adds the /shelf part and takes me to my audio shelf

**Comment by sensibleroommate** [comment]
![Screen Shot 2023-09-27 at 1 24 50 PM](https://github.com/houtianze/libby-audiobook-exporter-browser-extension/assets/145643032/1488ffdc-9f06-45f4-876e-f5e383868735)

**Comment by sensibleroommate** [comment]
This is what shows up

**Comment by pmyers389** [comment]
In the new version of this extension, the author changed the extension's methodology. Instead of the extension automatically activating on a page refresh you now need to click on the extension's icon in the toolbar. What follows is somewhat similar to the prior version. This obviates the need to refresh the screen to activate the extension.

**Comment by sensibleroommate** [comment]
![Screen Shot 2023-09-27 at 4 36 49 PM](https://github.com/houtianze/libby-audiobook-exporter-browser-extension/assets/145643032/e5fc0364-b88b-4853-831c-3486b1e7a99b)

Thanks, When i click on the extension, this is what opens up, I'm a bit confused as to how to download from here. Thanks

**Comment by sensibleroommate** [comment]
![Screen Shot 2023-09-27 at 4 38 44 PM](https://github.com/houtianze/libby-audiobook-exporter-browser-extension/assets/145643032/17db7a6a-02e1-479a-8043-3af6c06023df)

And then this message when I click on the extension. But no matter how much I refresh, the message doesn't change

**Comment by pmyers389** [comment]
Looks like I was wrong about the need to refresh. I tried another book and had to refresh a couple of times and didn't have a problem. I would suggest you try another book just to see if it works. If it works then maybe is something wrong with Classic Ghost Stories. If it doesn't work I would try removing and re-installing the extension. If that doesn't help I'm out of suggestions, other than asking the author for help.

**Comment by sensibleroommate** [comment]
Thanks, I'll try this and hopefully it will work, appreciate all your input!

**Comment by houtianze** [comment]
@pmyers389 Thanks for answering the questions. Yes, I did make a change and it now uses the popup page (after clicking on the extension icon). However, some (or most) of the time, the user still needs to refresh the page for the list of files to be populated. I think now I found another way to initiate the retrieval of the list of files, and if it does work, the refresh part won't be neccessary.

For now, from my experience, refreshing once of twice, I will be able to see the list of files and download.

**Comment by TrellisLore** [comment]
Read through all posts above, but no matter what I do it consistently get this message — done many refreshes, tried different books...checked permissions...any ideas?


![Screen Shot 2024-04-15 at 7 25 00 AM](https://github.com/houtianze/libby-audiobook-exporter-browser-extension/assets/167087360/1bfbf5ca-b255-43c7-9b51-ba20c209d361)

**Comment by pmyers389** [comment]
I am not having any issues with this extension. Are you using the version from the Chrome Web Store or the author's version from GitHub? I cannot remember which version I use but maybe if you try the version you are not using it might help.

**Comment by pmyers389** [comment]
Maybe test while disabling other extensions. Maybe try it on another version of Chrome such as Chrome portable. I'm using version 2.4 of the Libby extension with no issues. If you're having trouble on every book something is obviously radically wrong. In the Chrome extension manger you should see a Chrome security warning for this extension because they don't support it. If you don't see this message try again to install the author's version.

**Comment by TrellisLore** [comment]
I had seen the securty warning previously...I'd been using this for a few months when it suddenly stopped working the other day. I tried turning off other extension, no luck. I'm not particularly savvy in this field so I'm not sure what else I could do. I tried the similar extension for firefox and it's also not working. I've tried on two different computers, no luck...really not sure what is going on.

**Comment by TrellisLore** [comment]
I am getting an errors alert - but I don't know what this means. 
![Screenshot 2024-04-15 at 2 46 13 PM](https://github.com/houtianze/libby-audiobook-exporter-browser-extension/assets/167087360/22d3f574-08ba-4f2c-ab3f-106ca4c9dba7)

**Comment by NotKoalafied** [comment]
@TrellisLore there is another issue made for this specific problem. Link below.

https://github.com/houtianze/libby-audiobook-exporter-browser-extension/issues/12

**Comment by TrellisLore** [comment]
Thanks for the heads up, @NotKoalafied  — clearing cookies did the trick.
